kong/plugins/basic-auth/api.lua

```diff
@@ -52,5 +52,29 @@ return {
     DELETE = function(self, dao_factory)
       crud.delete(self.basicauth_credential, dao_factory.basicauth_credentials)
     end
+  },
+  ["/basic-auth/"] = {
+    GET = function(self, dao_factory)
+      crud.paginated_set(self, dao_factory.basicauth_credentials)
+    end
+  },
+  ["/basic-auth/:credential_id/consumer"] = {
+    before = function(self, dao_factory, helpers)
+      local credentials, err = dao_factory.basicauth_credentials:find_all { id = self.params.credential_id }
+
+      if err then
+        return helpers.yield_error(err)
+      elseif next(credentials) == nil then
+        return helpers.responses.send_HTTP_NOT_FOUND()
+      end
+
+      self.params.credential_key_or_id = nil
```

Comment:
This instruction is not needed since the parameter is called `credential_id` in this endpoint. Only if we follow my other suggestion of allowing `credential_username_or_id` does it become useful if we rename it properly (not `credential_key_*`).